### PR TITLE
[WIP] offlinemap: change to python3

### PR DESCRIPTION
**Pull request by ndowens**

### CODE REVIEW

**Comment by Piraty** [comment]
@pbui

**Comment by pbui** [comment]
Hmm... I will need to test it... my understanding is that offlineimap is not being ported to Python3 and thus using Python3 is unsupported: https://github.com/OfflineIMAP/offlineimap/issues/616

That said, I'm willing to test out these changes this weekend.

**Comment by ndowens** [comment]
Closing, thought it was just a config issue but I got pass that and yea this won’t work on py3

File: srcpkgs/python3-Pyro4/template [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+distfiles="${PYPI_SITE}/P/Pyro4/Pyro4-${version}.tar.gz"
+checksum=3897c0254046d4cb412a4d1a8f2f9c2c1c1ae643a24db07d0abdb51acdb8d7b5
+
+alternatives="
+ pyro4:pyro4-check-config:/usr/bin/pyro4-check-config2
+ pyro4:pyro4-flameserver:/usr/bin/pyro4-flameserver2
+ pyro4:pyro4-httpgateway:/usr/bin/pyro4-httpgateway2
+ pyro4:pyro4-ns:/usr/bin/pyro4-ns2
+ pyro4:pyro4-nsc:/usr/bin/pyro4-nsc2
+ pyro4:pyro4-test-echoserver:/usr/bin/pyro4-test-echoserver2"
```
Comment (sgn) [inline srcpkgs/python3-Pyro4/template]: Drop the alternates!
`build_style=python3-module` doesn't create versioned executable.

Even if they created versioned executable, 2 suffix indicated python2 version.

File: srcpkgs/python3-selectors2/template [hunk 2]
```diff
@@ -1,12 +1,12 @@
-# Template file for 'python-selectors2'
-pkgname=python-selectors2
+# Template file for 'python3-selectors2'
```
Comment (sgn) [inline srcpkgs/python3-selectors2/template]: selectors2 is back-ported version of selectors for python 2.x and 3.{0,4}
selectors is included in Core Python from python 3.5, drop it

Comment (sgn) [reply srcpkgs/python3-selectors2/template]: And also drop selectors dependencies in offlineimap.
I don't use offlineimap, I couldn't test it, though.

File: srcpkgs/python3-Pyro4/template (hunk 1 above)
Comment (ndowens) [reply srcpkgs/python3-Pyro4/template]: Didn’t actually add these ;) but ok

File: srcpkgs/python3-selectors2/template (hunk 2 above)
Comment (ndowens) [reply srcpkgs/python3-selectors2/template]: Currently offlineimap isn’t compatible with py3 but I think pbui is going to look at it; reason it is WIP atm

**Comment by ndowens** [comment]
Re-add python3-selectors as it caused build failure without.

**Comment by sgn** [comment]
> Re-add python3-selectors as it caused build failure without.

for rfc6555, we need this patch https://github.com/sethmlarson/rfc6555/commit/1a181b432312731f6742a5eb558dae4761d32361 https://github.com/sethmlarson/rfc6555/commit/1a181b432312731f6742a5eb558dae4761d32361.patch

File: srcpkgs/python3-Pyro4/template [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+build_style=python3-module
+pycompile_module="Pyro4"
+hostmakedepends="python3-setuptools"
+depends="python3-serpent>=1.27"
+short_desc="Distributed Object Middleware (IPC/RPC) for Python2"
+maintainer="pulux <pulux@pf4sh.de>"
+license="MIT"
+homepage="https://pythonhosted.org/Pyro4/"
+distfiles="${PYPI_SITE}/P/Pyro4/Pyro4-${version}.tar.gz"
+checksum=3897c0254046d4cb412a4d1a8f2f9c2c1c1ae643a24db07d0abdb51acdb8d7b5
```
Comment (sgn) [inline srcpkgs/python3-Pyro4/template]: Consider adding python-Pyro4 meta subpackage to remove its file.

**Comment by pbui** [comment]
Unfortunately, this doesn't work...  I get the following error when I try to sync one of my accounts:

```
IMAP4 protocol error: program error: <class 'TypeError'> - cannot use a bytes pattern on a string-like object.
```

**Pull request closed without merging**